tests/wp-unit/tests/include/Core/Posts/Metabox/DeletePostsByCategoryMetaboxTest.php

```diff
@@ -0,0 +1,726 @@
+			'days'           => 1,
+			'private'        => false,
+			'limit_to'       => false,
+			'force_delete'   => false,
+		);
+		$this->metabox->delete( $delete_options );
+
+		// Assert that post2 status moved to trash post1 status is publish.
+		$post1_status = get_post_status( $post1 );
+		$post2_status = get_post_status( $post2 );
+
+		$this->assertEquals( 'publish', $post1_status );
+		$this->assertEquals( 'trash', $post2_status );
+
+		// Assert that cat2 has no post and cat1 has one post.
+		$posts_in_cat1 = $this->get_posts_by_category( $cat1 );
+		$posts_in_cat2 = $this->get_posts_by_category( $cat2 );
+
+		$this->assertEquals( 1, count( $posts_in_cat1 ) );
+		$this->assertEquals( 0, count( $posts_in_cat2 ) );
+	}
+
+	/**
+	 * Create more than 100 posts and try to delete them in batches. Test at least 2 batches.
+	 */
+	public function test_for_delete_posts_as_batches() {
+		// Create a category.
+		$cat1 = $this->factory->category->create( array( 'name' => 'cat1' ) );
+		// Create 100 posts and assign the cat1 to all 100 posts.
+		$this->factory->post->create_many( 100, array(
+			'post_status' => 'publish',
+			'post_category' => array( $cat1 ),
+		) );
+
+		// Assert that cat1 has 100 posts.
+		$posts = $this->get_posts_by_category( $cat1 );
+		$this->assertEquals( 100, count( $posts ) );
+
+		// call our method. First batch limit_to 20 posts
+		$delete_options = array(
+			'post_type'      => array( 'post' ),
+			'selected_cats'  => array( $cat1 ),
+			'restrict'       => false,
+			'date_op'        => false,
+			'days'           => false,
+			'private'        => false,
+			'limit_to'       => 20,
+			'force_delete'   => false,
+		);
+		$this->metabox->delete( $delete_options );
+
+		// Assert that cat1 has 80 posts.
+		$posts = $this->get_posts_by_category( $cat1 );
+		$this->assertEquals( 80, count( $posts ) );
+
+		// call our method. Second batch limit_to 30 posts
+		$delete_options = array(
+			'post_type'      => array( 'post' ),
+			'selected_cats'  => array( $cat1 ),
+			'restrict'       => false,
+			'date_op'        => false,
+			'days'           => false,
+			'private'        => false,
+			'limit_to'       => 30,
+			'force_delete'   => false,
+		);
+		$this->metabox->delete( $delete_options );
+
+		// Assert that cat1 has 50 posts.
+		$posts = $this->get_posts_by_category( $cat1 );
+		$this->assertEquals( 50, count( $posts ) );
+	}
+
+	/**
+	 * Add tests to test deleting posts from All categories by custom post type.
+	 */
+	public function test_for_deleting_posts_from_all_categories_custom_post_type() {
+		$taxonomy  = 'category';
+		$post_type = 'movie';
+
+		// Create two terms.
+		$term1 = $this->factory->term->create( array( 'name' => 'term1', 'taxonomy' => $taxonomy ) );
+		$term2 = $this->factory->term->create( array( 'name' => 'term2', 'taxonomy' => $taxonomy ) );
+
+		// Create two posts and assign the term.
+		$post1 = $this->factory->post->create( array( 'post_title' => 'post1', 'post_type' => $post_type ) );
```

Comment:
You need to register the post type before using it. Otherwise it will default to the built-in 'post' post type. That's what is happening here.

Reply:
Fixed